larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc

```diff
@@ -0,0 +1,394 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc
+ *
+ *  @brief  Implementation of the lar adaptive boost decision tree class.
+ *
+ *  $Log: $
+ */
+
+#include "Helpers/XmlHelper.h"
+
+#include "larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h"
+
+namespace lar_content
+{
+
+AdaBoostDecisionTree::AdaBoostDecisionTree() :
+    m_pStrongClassifier(nullptr)
+{
+}
+
```

Comment:
Destructor needed to delete `m_pStrongClassifier`. Note that deleting `nullptr` is fine, so don't need to check before delete, but otherwise need to check before any attempt to deference.

Then need to add copy constructor and copy assignment operator, as the system defaults would e.g. give pointer to rhs-owned strong classifier to the lhs, which could proceed to delete it.

See https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming), now rule of five with C++11, although we probably don't adhere to the rule of five everywhere yet.